### PR TITLE
Feature/add unit tests

### LINKED ISSUE #1790: Increase test coverage
### Description
Currently our test coverage measured by Codecov is pointing to a 35% code coverage. The goal is to improve upon this metric and add the necessary unit tests.

We are aiming for a 80%+ code coverage

### PR DESCRIPTION
**Pull request by joaoReigota1**
Closes #1790

**Proposed Changes**

- Increase unit test coverage

### CODE REVIEW

File: pkg/parser/yaml/parser_test.go [hunk 1]
```diff
@@ -19,17 +19,33 @@ func TestParser_SupportedExtensions(t *testing.T) {
...
+`, `
+---
+- name: Create an empty bucket2
+  amazon.aws.aws_s3:
+    bucket: mybucket
+    mode: create
+    permission: authenticated-read
+`, `
+-name: teste
+-name: teste2
```
Comment (ruigomescx) [inline pkg/parser/yaml/parser_test.go:37]: change the names of "teste" and "teste"

File: internal/console/generate_id_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	return out, err
+}
+
+func captureCommandOutput(cmd *cobra.Command, args []string) (string, error) {
+	if len(args) > 0 {
+		cmd.SetArgs(args)
+	}
+
+	return captureOutput(cmd.Execute)
+}
```
Comment (ruigomescx) [inline internal/console/generate_id_test.go:45]: Move this code to a helper file, maybe inside /tests

File: pkg/builder/parser/tag/tag_parser.go [hunk 3]
```diff
@@ -180,7 +180,7 @@ func parseArgs(sc *scanner.Scanner) (map[string]interface{}, error) {
 		}
 		result[name] = value
 		next := sc.Next()
-		if next == ')' {
+		if next == ']' {
```
Comment (rogeriopeixotocx) [inline pkg/builder/parser/tag/tag_parser.go:183]: what happened here?

File: internal/console/version_test.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestVersionCommand(t *testing.T) {
+	t.Run("Tests if prints current version", func(t *testing.T) {
+		out, err := captureCommandOutput(versionCmd, nil)
+
+		require.NoError(t, err)
+		require.Equal(t, "Keeping Infrastructure as Code Secure v1.1.1\n", out)
```
Comment (ruigomescx) [inline internal/console/version_test.go:14]: The version string should be converted into a constant (this goes for the actual implementation of the getVersion() as well)

File: pkg/engine/query/source_test.go [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+	}
+}
+
+func getCurrentDirName(path string) string {
+	dirs := strings.Split(path, string(os.PathSeparator))
+	if dirs[len(dirs)-1] == "" && len(dirs) > 1 {
+		return dirs[len(dirs)-2]
+	}
+	return dirs[len(dirs)-1]
+}
```
Comment (ruigomescx) [inline pkg/engine/query/source_test.go:215]: functions being used in `pkg/engine/inspector_test.go`. Make these functions inside a common helper file

File: pkg/engine/query/source.go [hunk 6]
```diff
@@ -36,7 +36,7 @@ func GetPathToLibrary(platform, relativeBasePath string) string {
 
 	if strings.Contains(platform, "ansible") {
 		return filepath.FromSlash(libraryPath + "/ansible/" + LibraryFileName)
-	} else if strings.Contains(platform, "cloudformation") {
+	} else if strings.Contains(platform, "cloudFormation") {
```
Comment (rogeriopeixotocx) [inline pkg/engine/query/source.go:39]: should we do an ignore case check in those decision structures?

File: pkg/kics/service_test.go [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+		})
+		t.Run(fmt.Sprintf(tt.name+"_start_scan"), func(t *testing.T) {
+			if err := s.StartScan(tt.args.ctx, tt.args.scanID); (err != nil) != tt.wantErr {
+				t.Errorf("Service.StartScan() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func createParserSoureceProvider(path string) (*parser.Parser, *source.FileSystemSourceProvider) {
```
Comment (ruigomescx) [inline pkg/kics/service_test.go:103]: typo in createParser**Sourece**Provider -> Source

File: pkg/model/summary_test.go [hunk 8]
```diff
@@ -0,0 +1,87 @@
+package model
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateSummary(t *testing.T) {
+	vulnerabilitys := []Vulnerability{
```
Comment (ruigomescx) [inline pkg/model/summary_test.go:10]: **vulnerabilities***

File: pkg/parser/terraform/converter/default_test.go [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+		t.Fatal("parse bytes:", err)
+	}
+	var v map[string]interface{}
+	err = json.Unmarshal([]byte(expected), &v)
+	if err != nil {
+		t.Fatal("parse bytes:", err)
+	}
+
+	require.Len(t, body, len(v))
+	require.Equal(t, v["data"].(map[string]interface{})["terraform_remote_state"].(map[string]interface{})["remote"].(map[string]interface{})["backend"], // nolint
```
Comment (rogeriopeixotocx) [inline pkg/parser/terraform/converter/default_test.go:238]: maybe a comment to explain this line for the future

File: test/fixtures/all_auth_users_get_read_access/test/negative.tf [hunk 10]
```diff
@@ -0,0 +1,9 @@
+resource "aws_s3_bucket" "b" {
+  bucket = "my-tf-test-bucket"
+  acl    = "private"
+
+  tags = {
+    Name        = "My bucket"
+    Environment = "Dev"
+  }
+}
```
Comment (rogeriopeixotocx) [inline test/fixtures/all_auth_users_get_read_access/test/negative.tf:9]: missing EOF newline

File: test/fixtures/all_auth_users_get_read_access/test/positive.tf [hunk 11]
```diff
@@ -0,0 +1,9 @@
+resource "aws_s3_bucket" "b" {
+  bucket = "my-tf-test-bucket"
+  acl    = "authenticated-read"
+
+  tags = {
+    Name        = "My bucket"
+    Environment = "Dev"
+  }
+}
```
Comment (rogeriopeixotocx) [inline test/fixtures/all_auth_users_get_read_access/test/positive.tf:9]: missing EOF newline

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

File: pkg/parser/terraform/converter/default_test.go [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+		body["data"].(model.Document)["terraform_remote_state"].(model.Document)["remote"].(model.Document)["backend"])
```
Comment (ruigomescx) [inline pkg/parser/terraform/converter/default_test.go:239]: can we refactor this part to make a more comprehensible test case?

**Review by ruigomescx — CHANGES_REQUESTED** [review]
Made some comments and suggestions. Overall looks pretty good, I would say that the only thing missing might be having more comments throughout the tests to make it easier to understand the tests being made. We could at least give a short description on each test, and inside each test when applicable

File: internal/console/version_test.go (hunk 4 above)
Comment (felipe-avelar) [reply internal/console/version_test.go:14]: Should correct version.go in this PR?

File: pkg/engine/query/source.go (hunk 6 above)
Comment (joaoReigota1) [reply pkg/engine/query/source.go:39]: resolved

File: pkg/engine/query/source_test.go (hunk 5 above)
Comment (joaoReigota1) [reply pkg/engine/query/source_test.go]: resolved

File: internal/console/generate_id_test.go [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+
+func TestGenerateIDCommand(t *testing.T) {
+	t.Run("Tests if generates a valid uuid", func(t *testing.T) {
+		validUUID := regexp.MustCompile(`(?i)^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$`)
```
[the remark this answers is not in the document]
Comment (felipe-avelar) [reply internal/console/generate_id_test.go]: Added a constant to test/helpers.go with the rule and changed files with the regex to use the constant

File: pkg/parser/terraform/converter/default_test.go (hunk 12 above)
Comment (joaoReigota1) [reply pkg/parser/terraform/converter/default_test.go:239]: resolved

File: internal/console/generate_id_test.go (hunk 2 above)
Comment (felipe-avelar) [reply internal/console/generate_id_test.go]: Moved to test/helpers.go

File: pkg/engine/query/source_test.go (hunk 5 above)
Comment (felipe-avelar) [reply pkg/engine/query/source_test.go]: Moved to test/helpers.go

File: internal/console/version_test.go (hunk 4 above)
Comment (felipe-avelar) [reply internal/console/version_test.go]: Changed to use `currentVersion` constants in version.go and version_test.go

File: pkg/engine/query/source.go [hunk 14]
```diff
@@ -34,15 +34,15 @@ const (
 func GetPathToLibrary(platform, relativeBasePath string) string {
 	libraryPath := filepath.Join(relativeBasePath, LibrariesBasePath)
 
-	if strings.Contains(platform, "ansible") {
+	if strings.Contains(strings.ToUpper(platform), strings.ToUpper("ansible")) {
```
Comment (rogeriopeixotocx) [inline pkg/engine/query/source.go:37]: looking good

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**